hw1/hw-Ivaniv.js

```diff
@@ -0,0 +1,100 @@
+//task1
+
+function copyString(string, count) {
+    return string.repeat(count);
+}
+
+console.warn(copyString('a', 5));
+
+
+//task2
+
+function searchSubString(subString) {
+    if (!subString) return;
+    return subString.toLowerCase().search(/^if/) >= 0 ? true : false;
+}
+
+console.warn(searchSubString('if gjgjhvhjgv'));
+
+
+//task3
+function checkLastDigits(first, second) {   
+    return first.toString().split('').pop() == second.toString().split('').pop();
+}
+
+console.warn(checkLastDigits(12, 32541));
+
+//task4
```

Comment:
Have you changed anything with it at all? I don't like how it looks like.
first of all, function which is named sum returns not the sum but all the results.
secondly, you are validating the argument in further functions (searchSameNumbers, splitArray) but it is enough to check the argument once before you start doing any operations with it.
third, The naming of splitArray is not appropriate. It should be getArrayHalfsEquality because it's not obvious what this function does.

If you are making each step as a separate function then DON'T put all the steps inside function called sum. If function is called sum it is assumed that it ONLY returns a sum of something and doesn't do anything else.

I would suggest to create a function `getNumberStats(numberOf4Digits)` which does the next steps:
1. validates the argument (checks if it is number, if its length is 4 etc. )
2. If validation passed create an array from argument.
3. create a `const` which will be a result of sum function: `const numberSum = sum(numberArr) // sum ONLY returns asum of digits`
4. create a `const` which will be a result of searchSameNumber: `const hasSameNumbers = searchSameNumbers(numberArray) // searchSameNumbers DOESN'T validate the value`
5. create a `const` which will be a result of getArrayHalfsEquality (which right now called splitArray): `const isHalfsEqual = getArrayHalfsEquality(numberArray)`
6.return resulting object with stats: `return {numberSum, hasSameNumbers, isHalfsEqual}`

Please, make these changes. In case of questions ask me in skype